util/feeding.py

```diff
@@ -132,10 +132,13 @@ def batch_fn(wav_filenames, features, features_len, transcripts):
 
     dataset = (tf.data.Dataset.from_generator(generate_values,
                                               output_types=(tf.string, (tf.int64, tf.int32, tf.int64)))
-                              .map(process_fn, num_parallel_calls=tf.data.experimental.AUTOTUNE)
-                              .cache(cache_path)
-                              .window(batch_size, drop_remainder=True).flat_map(batch_fn)
-                              .prefetch(num_gpus))
+                              .map(process_fn, num_parallel_calls=tf.data.experimental.AUTOTUNE))
+
+    if cache_path:
```

Comment:
You need to check for `None` explicitly (`if cache_path is not None`) because in Python empty-strings are "falsey". And if `cache_path` is an empty string we still want to cache to memory. Alternatively if the tri-state is too confusing feel free to add a separate parameter.